internal/exec/exec.go

```diff
@@ -51,7 +51,9 @@ func (r *Request) Execute(ctx context.Context, s *resolvable.Schema, op *query.O
 	}()
 
 	if err := ctx.Err(); err != nil {
-		return nil, []*errors.QueryError{errors.Errorf("%s", err)}
+		//If context has either been cancelled or timed out we still may want to return the features that have finished
+		// TODO properly attribute mark which features have timedout in the error field
```

Comment:
is there a potential race condition where the deadline is exceeded and before we've responded a field resolves and writes an error to the output so we got the impression that a field failed for some reason but actually it was because the deadline exceeded?